src/main/java/org/quiltmc/loader/impl/QuiltLoaderImpl.java

```diff
@@ -117,7 +119,28 @@ public void setGameProvider(GameProvider provider) {
 
 	private void setGameDir(Path gameDir) {
 		this.gameDir = gameDir;
-		this.configDir = gameDir.resolve("config");
+		String configDir = System.getProperty(SystemProperties.CONFIG_DIRECTORY);
+		this.configDir = resolveIfNotAbsolute(gameDir, configDir == null || configDir.isEmpty()  ? "config" : configDir);
+		refreshModsDir(gameDir);
+	}
+
+	private void refreshModsDir(Path gameDir) {
+		String modsSubDir = System.getProperty(SystemProperties.MODS_DIRECTORY);
+		this.modsDir = resolveIfNotAbsolute(gameDir, modsSubDir == null || modsSubDir.isEmpty() ? "mods" : modsSubDir);
+	}
+
+	/**
+	 * If the passed string toResolve is an absolute path, it will be
+	 * returned as a path. If it is not absolute, it will be returned
+	 * as a subdirectory of the passed rootDir.
+	 *
+	 * @param rootDir The parent directory to use if toResolve is not absolute
+	 * @param toResolve The subdirectory/absolute path to be interpreted
+	 * @return Either a path leading to a subdirectory of rootDir, or an absolute path
+	 */
+	private Path resolveIfNotAbsolute(Path rootDir, String toResolve) {
+		Path abs = Paths.get(toResolve);
+		return abs.isAbsolute() ? abs : rootDir.resolve(toResolve);
```

Comment:
From the documentation of `Path.resolve(Path other)`:

````java
/** If the {@code other} parameter is an {@link #isAbsolute() absolute}
 * path then this method trivially returns {@code other}. */
````

As such you can replace this entire method with just a call to `Path.resolve(String other)`, since that method *also* inherits the same behaviour. (And it obtains a `Path` using basically the same logic as `Paths.get(String)`)

------------

This *also* means my [earlier comment](https://github.com/QuiltMC/quilt-loader/pull/20#issuecomment-855326673) was wrong - the original code already supported absolute paths, and [this change](https://github.com/QuiltMC/quilt-loader/pull/20/commits/72be98c6e64e2c527f87e0b01a0792bed2e2d389#diff-96a52e7f80ce90440904f48ddd2fc4209f1875799268a24aff53471f8f06ef3cL122-R143) wasn't necessary. Sorry :/

(However the name change of the system property is probably a good thing - although I guess we'll have to document that the property can be used for both absolute and sub-paths)?

Reply:
I noticed this in others' comments a while after the last commit, and didn't have much luck finding documentation to back it up, now I know to revert this change. Thanks